packages/contracts/src/EmailAccountRecovery.sol

```diff
@@ -146,7 +171,13 @@ abstract contract EmailAccountRecovery {
         EmailAuthMsg memory emailAuthMsg,
         uint templateIdx
     ) external {
+        address recoveredAccount = extractRecoveredAccountFromAcceptanceSubject(
+            emailAuthMsg.subjectParams,
+            templateIdx
+        );
+        require(recoveredAccount != address(0), "invalid account in email");
         address guardian = computeEmailAuthAddress(
+            recoveredAccount,
             emailAuthMsg.proof.accountSalt
         );
         require(
```

Comment:
Just want to highlight this as a discussion point and not necessarily suggesting this as a change.

It is possible that an account could have multiple validators that need recovery configured for, and they may also want to use the same email address for multiple validators. In this case, this would not be possible with the current contracts as adding the same guardian again would revert in `handleAcceptance`. 

This may be too much of a scope increase for the time we have, and a simple work around would be to use a different guardian or email address for this scenario. I'm not massively in favour of this because of these reasons but wanted to highlight it while we're changing the contracts. Here was a [change](https://github.com/zkemail/email-recovery/blob/6b6e31df337243bfa593fa08bda0c7a8ee522039/src/EmailAccountRecoveryNew.sol#L158-L171) I played around with when thinking about how you would allow the same guardian more than once per account

Reply:
Hmmm actually thinking about it some more, a change along these lines might be necessary if we want to go with Rhinestone's advice and clear all module state when a recovery module is uninstalled. 

Under the assumption that recovery state is cleared completely, handleAcceptance can be called for the same account and guardian email address combo more than once. Which with the current logic, would fail. So no one would be able to uninstall a recovery module and then reinstall it again and get recovery working.

Which has then got me thinking about the implications of clearing/not clearing `EmailAuth` state. Will have to think some more about that

Reply:
Thank you for your comment.
I agree with this [change](https://github.com/zkemail/email-recovery/blob/6b6e31df337243bfa593fa08bda0c7a8ee522039/src/EmailAccountRecoveryNew.sol#L158-L171).
Even if a wallet owner uninstalls a module for email account recovery, a relayer cannot detect it since its implementation is agnostic to the module implementation, and it will not issue a new account code for the same combination of the account and the guardian's email address.
However, it requires the `acceptGuardian`, `processRecovery`, and `completeRecovery` functions in the module to assert that the module is still installed by the account.
Otherwise, it can be a DoS attack vector against a relayer.
(For example, a malicious user uninstalls the module but starts recovery, but the final `completeRecovery` should fail as the function to rotate the owner does not work if the module is uninstalled.)

@JohnGuilding Could you add those assertions to your module?
@wshino Could you add this [change](https://github.com/zkemail/email-recovery/blob/6b6e31df337243bfa593fa08bda0c7a8ee522039/src/EmailAccountRecoveryNew.sol#L158-L171)?

Reply:
> implications of clearing/not clearing EmailAuth state

I think we don't have to clear `EmailAuth` state when the module is uninstalled because the module will require the `EmailAuth` to hold the same configuration states such as `accountSalt`, `module`, `subjectTemplates`, and so on.
Also, I think it is better to maintain `authedHash` states to prevent unexpected replay/DoS attacks.

Reply:
> However, it requires the acceptGuardian, processRecovery, and completeRecovery functions in the module to assert that the module is still installed by the account.
Otherwise, it can be a DoS attack vector against a relayer.

I see, so in this scenario, someone could call `handleRecovery` (or other functions) via the relayer but the module is not installed, so the relayer would have to pay for transactions that would always fail?

We already do the following check when configuring recovery & updating recovery config. Will add this change to `acceptGuardian` & `processRecovery`

```solidity
        bool isInitialized = IModule(recoveryConfig.recoveryModule).isInitialized(account);
        if (!isInitialized) {
            revert RecoveryModuleNotInstalled();
        }
```

> I think we don't have to clear EmailAuth state when the module is uninstalled because the module will require the EmailAuth to hold the same configuration states such as accountSalt, module, and subjectTemplates.
Also, I think it is better to maintain authedHash states to prevent unexpected replay/DoS attacks.

I agree. I don't see a need to do so

Reply:
> but the module is not installed, so the relayer would have to pay for transactions that would always fail?
Yes.
> We already do the following check when configuring recovery & updating recovery config.
This check seems to be perfect!
> Will add this change to acceptGuardian & processRecovery
Thank you very much!

Reply:
The following check is still in this function
```solidity
        require(
            address(guardian).code.length == 0,
            "guardian is already deployed"
        );
```

This needs removing otherwise the following block will never be reached:
```solidity
if (guardian.code.length == 0) {
...
```

Same in `handleRecovery`